### PR TITLE
fix(cli): fix empty response handling in call cmd

### PR DESCRIPTION
**Pull request by thsig**
**What this PR does / why we need it**:

Before this fix, the `call` command would throw an exception when calling the service resulted in an empty response.

### CODE REVIEW

File: garden-service/src/commands/call.ts [hunk 1]
```diff
@@ -168,34 +169,39 @@ export class CallCommand extends Command<Args> {
...
 
     try {
       res = await req
       entry.setSuccess()
       statusText = getStatusText(res.statusCode)
       log.info(chalk.green(`${res.statusCode} ${statusText}\n`))
+
+      output = res.body
+
+      if (res.headers["content-type"] === "application/json") {
```
Comment (eysi09) [inline garden-service/src/commands/call.ts]: Couldn't `"content-type"` be capitalised?

File: garden-service/src/commands/call.ts [hunk 2]
```diff
@@ -204,8 +210,9 @@ export class CallCommand extends Command<Args> {
         url,
         response: {
           data: output,
-          headers: res.headers,
-          status: res.statusCode,
+          error,
+          headers: res ? res.headers : null,
+          status: res ? res.statusCode : 204,
```
Comment (eysi09) [inline garden-service/src/commands/call.ts]: Is it valid to just default to `204` if there's not status code?

Comment (thsig) [reply garden-service/src/commands/call.ts]: The idea was to use the status code for "No Content" (or "empty response") here.

I.e. the status code for "we received a response, but it was empty".

File: garden-service/src/commands/call.ts (hunk 1 above)
Comment (eysi09) [reply garden-service/src/commands/call.ts]: It could still be lower cased.

Comment (edvald) [reply garden-service/src/commands/call.ts]: I believe the library lowercases all response headers

Comment (eysi09) [reply garden-service/src/commands/call.ts]: Ok nice, couldn't find it in their docs.

Comment (eysi09) [reply garden-service/src/commands/call.ts]: @thsig  ☝️

File: garden-service/src/commands/call.ts (hunk 2 above)
Comment (edvald) [reply garden-service/src/commands/call.ts]: Seems a little confusing to me actually, I'd rather just allow a null value here to indicate that there was no status code received.

**Pull request merged**